src/components/common/PeopleAndRoles/ListPeopleAndRoles.vue

```diff
@@ -45,227 +45,325 @@
-              :activeIndex="activeIndex"
-              @addEdit="emitAddEdit($event)"
-              @remove="emitRemove($event)"
-              @reset="emitReset()"
-            />
-          </div>
-        </v-expand-transition>
-
-        <!-- normal table row -->
-        <template v-if="!renderOrgPersonForm || (index != activeIndex)">
-          <!-- Name + Badge -->
-          <v-col class="pr-2" cols="12" sm="3">
-            <v-row no-gutters>
-              <v-col cols="1" class="mt-n1 ml-n1 mr-3 badges" :class="{ 'removed': wasRemoved(orgPerson)}">
-                <v-icon color="gray9" v-if="isPartyTypePerson(orgPerson)">mdi-account</v-icon>
-                <v-icon color="gray9" v-if="isPartyTypeOrg(orgPerson)">mdi-domain</v-icon>
-              </v-col>
-              <v-col class="overflow-hidden">
-                <p class="people-roles-title mb-1" :class="{ 'removed': wasRemoved(orgPerson)}">
-                  {{ formatName(orgPerson) }}
-                </p>
-                <p class="info-text mb-1 people-roles-email" :class="{ 'removed': wasRemoved(orgPerson)}">
-                  {{ orgPerson.officer.email }}
-                </p>
-                <p
-                  v-if="orgPerson.officer.identifier"
-                  class="info-text mb-1 people-roles-inc-number"
-                  :class="{ 'removed': wasRemoved(orgPerson)}">
-                  Incorporation Number: {{orgPerson.officer.identifier}}
-                </p>
-
-                <div v-if="!isSummaryView && orgPerson.actions">
-                  <v-chip v-if="wasRemoved(orgPerson)" x-small label color="#grey lighten-2">
-                    {{ orgPerson.actions[0] }}
-                  </v-chip>
-                  <v-chip
-                    v-else
-                    v-for="(action, i) in orgPerson.actions"
-                    :key="`action-chip-${i}`"
-                    x-small label color="primary" text-color="white">
-                    {{ action }}
-                  </v-chip>
-                </div>
-              </v-col>
-            </v-row>
-          </v-col>
-
-          <!-- Mailing Address -->
-          <v-col cols="12" :sm="isSummaryView ? 4 : 3" :class="{ 'removed': wasRemoved(orgPerson)}">
-            <MailingAddress class="peoples-roles-mailing-address" :address="orgPerson.mailingAddress" />
-          </v-col>
-
-          <!-- Delivery Address -->
-          <v-col cols="12" sm="3" :class="{ 'removed': wasRemoved(orgPerson)}">
-            <template
-              v-if="hasRoleDirector(orgPerson) || hasRoleProprietor(orgPerson) || hasRolePartner(orgPerson)"
-            >
-              <template v-if="IsSame(orgPerson.mailingAddress, orgPerson.deliveryAddress, ['id'])">
-                <span class="peoples-roles-delivery-address info-text">Same as Mailing Address</span>
+      <template v-for="(orgPerson, index) in currentPeopleAndRoles">
+        <v-row
+          v-if="!(wasReplaced(orgPerson) && wasRemoved(orgPerson))"
```

Comment:
It's importand to iterate (v-for) over the entire list so that indexes are correct.

Specifically, a replaced-removed item is not displayed but is still in the list so it takes up an index in this loop.